src/AlgebraicGeometry/Surfaces/EnriquesAut.jl

```diff
@@ -0,0 +1,944 @@
+  @vprint :EnriquesAuto 2 "computing stabilizer "
+  stab_Dminus, inc_stab = stabilizer(GSm, inc_Dminus)
+  @vprint :EnriquesAuto 2 "done\n"
+  Dminus_perp, inc_Dminus_perp = orthogonal_submodule(DSm, Dminus)
+  res, inc = restrict_automorphism_group(stab_Dminus, inc_Dminus_perp)
+  res2, inc2 = restrict_automorphism_group(stab_Dminus, inc_Dminus)
+  @vprint :EnriquesAuto 2 "computing kernel "
+  Gminus, inc_Gminus = kernel(inc)
+  @vprintln :EnriquesAuto 2 "done"
+  ECtx.orderGbar = order(Gminus)
+
+  # compute Gplus
+  Q = orthogonal_submodule(L26,SY)
+  DQ = discriminant_group(Q)
+  glue_SY_Q,inc1,inc2 = glue_map(L26,SY, Q)
+  DO = codomain(glue_SY_Q)
+
+  @vprint :EnriquesAuto 2 "preparing membership test\n"
+  DSY = discriminant_group(SY)
+  ODSY = orthogonal_group(DSY)
+  tmp = [preimage(phiSm, i) for i in small_generating_set(Gminus)]
+  tmp_hom = [hom(DO,DO, [DO(lift(j)*i) for j in gens(DO)]) for i in tmp]
+  gens_Gplus = elem_type(ODSY)[ODSY(inv(inc1)*glue_SY_Q*i*inv(glue_SY_Q)*inc1) for i in tmp_hom]
+  Gplus,_ = sub(ODSY, gens_Gplus)
+  @assert order(Gplus) == order(Gminus)
+  # iso_Gplus_minus = hom(Gplus, Gminus, gens(Gplus), small_generating_set(Gminus))
+
+  Dplus = codomain(phi)
+  ODplus = orthogonal_group(Dplus)
+  # turn this into a mod 2 matrix group
+  # with respect to the basis of SY mod 2
+  B = 1//2*basis_matrix(SY)
+  gens_Gplus_mat = FqMatrix[]
+  for g in gens(Gplus)
+    g2 = reduce(vcat,[change_base_ring(GF(2),solve(B,matrix(QQ,1,26, lift(g(DSY(vec(B[i,:])))));side=:left))  for i in 1:10])
+    push!(gens_Gplus_mat, g2)
+  end
+  Gplus_mat = matrix_group(GF(2),10, gens_Gplus_mat)
+  ECtx.Gplus_mat = Gplus_mat
+
+  snf_Dplus, i_snf = snf(Dplus)
+  gens_Dplus = [change_base_ring(GF(2),solve(basis_matrix(SY),matrix(QQ,1,26,2*lift(inc_Dplus(i_snf(i))));side=:left)) for i in gens(snf_Dplus)]
+
+  gens_Dplus = reduce(vcat, gens_Dplus)
+  ECtx.Dplus = gens_Dplus
+
+
+  Dplus_perp, _ = orthogonal_submodule(DSY, Dplus)
+  Dplus_perp,_ = snf(Dplus_perp)
+
+  # needed for hash
+  ECtx.Dplus_perp = reduce(vcat,[change_base_ring(GF(2),2*solve(basis_matrix(SY),matrix(QQ,1,26,lift(i));side=:left)) for i in gens(Dplus_perp)])
+
+
+  @vprint :EnriquesAuto 2 "done\n"
+  ECtx.membership_test = membership_test_as_group
+  #upper_bound = mass(ECtx)*length(initial_automorphisms(ECtx))
+  return ECtx
+end
+  
+@doc raw"""
+    EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat) -> EnriquesBorcherdsCtx 
+  
+Return a context object for Borcherds' method for Enriques surfaces. 
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surfaces and ``\epsilon`` the Enriques involution. 
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+"""
+function EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat)
+  @req issubset(SY2, SX) "SY2 must be a sublattice of SX"
+  a = rescale(SY2,1//2)
+  @req is_unimodular(a) && is_even(a) && signature(a) == (1, 0, 9) "SY2 must be isomorphic to E10(2)"
```

Comment:
```suggestion
  @req is_unimodular(a) && is_even(a) && signature_tuple(a) == (1, 0, 9) "SY2 must be isomorphic to E10(2)"
```
Did `signature(a)` work on your side ?